### ISSUE TITLE
should we handle uint256::max as infinite allowance

### ISSUE THREAD

**Issue by ermalkaleci** [AcalaNetwork/predeploy-contracts#86]
https://github.com/OpenZeppelin/openzeppelin-contracts/blob/c72281ea4506e92434ed9ac9c87a9cb13a55c6ce/contracts/token/ERC20/ERC20.sol#L336

**Comment by zjb0807** [comment]
https://github.com/OpenZeppelin/openzeppelin-contracts/pull/3085/files

```
     * NOTE: If `value` is the maximum `uint256`, the allowance is not updated on
     * `transferFrom`. This is semantically equivalent to an infinite approval.
```
It's good for me.

**Comment by ermalkaleci** [comment]
closed by #85
